app/models/internal/scheduling/Job.java

```diff
@@ -46,13 +45,6 @@
      */
     Schedule getSchedule();
 
-    /**
-     * Returns the last time the job was executed for (or {@code empty()} if it has never run).
-     *
-     * @return The time.
-     */
-    Optional<Instant> getLastRun();
```

Comment:
Why are we removing this?

-- the idea is that you need to reload the job from the repository when its state has changed.

Reply:
As discussed in person: this method now lives on `JobRepository`. @cwbriones is updating his diff to use that method instead of this one.

(I feel fairly strongly that we don't want to introduce an extra layer of caching here: the last run should be cached in the `JobRepository` if at all, I opine, until we see need for another layer.)